org-roam-db.el

```diff
@@ -234,6 +234,19 @@ If FILE is nil, clear the current buffer."
                       :where (= file $s1)]
                      file))
 
+;;;;; Protection against unique-ID errors
+(define-error 'emacsql-constraint "SQL constraint violation")
+(defun org-roam--protected-node-insert (query vec)
+  (catch 'sqlerror
+    (condition-case err
+        (org-roam-db-query query vec)
+      (emacsql-constraint
+       (throw 'sqlerror
```

Comment:
Doesn't seem useful to catch the error immediately in the same function. Could probably remove the catch and throw clauses, `condition-case` is enough

Reply:
I was having problems where `condition-case` was still aborting the control flow. I think I just figured out how this works.